src/cli/find-cache-file.js

```diff
@@ -4,16 +4,26 @@ const os = require("os");
 const path = require("path");
 const findCacheDir = require("find-cache-dir");
 
+function findDefaultCacheFile() {
+  const cacheDir =
+    findCacheDir({ name: "prettier", create: true }) || os.tmpdir();
+  const cacheFilePath = path.join(cacheDir, ".prettier-cache");
+  return cacheFilePath;
+}
+
 /**
  * Find default cache file (`./node_modules/.cache/prettier/.prettier-cache`) using https://github.com/avajs/find-cache-dir
  *
+ * @param {string | undefined} cacheLocation
  * @returns {string}
  */
-function findCacheFile() {
-  const cacheDir =
-    findCacheDir({ name: "prettier", create: true }) || os.tmpdir();
-  const cacheFilePath = path.join(cacheDir, ".prettier-cache");
-  return cacheFilePath;
+function findCacheFile(cacheLocation) {
+  if (!cacheLocation) {
+    return findDefaultCacheFile();
+  }
+  const cwd = process.cwd();
+  const normalized = path.normalize(cacheLocation);
+  return path.join(cwd, normalized);
```

Comment:
I don't think it's a good idea to use the input path directly, people may pass wrong path by accident,  if path exists, we should verify

- It's a file not a directory
- Verify it's a valid Prettier cache file (maybe add a mark to the cache when save cache)

Reply:
> It's a file not a directory

done! [9789e04](https://github.com/prettier/prettier/pull/13019/commits/9789e04e017152fafa72a38e80807320d853e34c)

> Verify it's a valid Prettier cache file (maybe add a mark to the cache when save cache)

How?

Reply:
> How?

I'm not sure how the cache file is created, but is it a JSON file? There should be a way to detect the file is created by Prettier.

Reply:
That file is a JSON file, but it is almost always huge. There is an overhead in making sure it is in the correct format. I don't think its validation is worth the overhead.

Reply:
Personally, I hope it's safer, I'll be pissed if `prettier --cache --cache-location index.js src`(forgot cache file path) saves cache to my `index.js`

Reply:
[afcbd3c](https://github.com/prettier/prettier/pull/13019/commits/afcbd3ca39296a3bc707aa8b9d7475bda27e3816)

I've added JSON validation. But it consumes a lot of memory. We should use Stream in the future to implement JSON checks with low memory usage.

Reply:
Is there a specially key inside that we can confirm it's a Prettier cache file?

Reply:
I don't know, need to read the [file-entry-cache](https://github.com/royriojas/file-entry-cache/blob/master/cache.js) implementation

Reply:
I want to break up the PR because one big PR is exhausting. This is listed in the TODO of this PR.

Reply:
Okay?